internal/bminventory/inventory.go

```diff
@@ -497,7 +497,14 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		cpuArchitecture = common.MultiCPUArchitecture
 		// (MGMT-11859) Additional check here ensures that the customer cannot just guess a version
 		//              with multiarch in order to get access to that release payload.
-		if err = b.versionsHandler.ValidateAccessToMultiarch(ctx, b.authzHandler); err != nil {
+		var multiarchAllowed bool
+		multiarchAllowed, err = b.authzHandler.HasOrgBasedCapability(ctx, ocm.MultiarchCapabilityName)
+		if err != nil {
+			err = common.NewApiError(http.StatusInternalServerError, fmt.Errorf("error getting user %s capability, error: %w", ocm.MultiarchCapabilityName, err))
```

Comment:
It is not advisable to expose the details of the auth capability we use to the user. I would change the error message here to something more generic .

Reply:
Also I do think that a utility function that hide this code will be more readable. The former ValidateAccessToMultiarch was very clear. If you like a different error handling you can inject an error message but personally I don't see why we need a different errors here and in versionhandler.

Reply:
Here you're probably right, we likely don't need different errors, but in the version handler we have different behavior for error vs if multi arch is not available. This is what made sharing the implementation rather strange.